### PR TITLE
[GHF] Fix pushDate handling

### PR DESCRIPTION
**Pull request by malfet**
Merge commits does not have a merge date, which is also clear from [GraphQL schema](https://docs.github.com/en/graphql/reference/objects#commit).
Modify return signature of `GitHubPR.last_pushed_at`, print warning when one can not be queried and add regression test.

### CODE REVIEW

File: .github/scripts/test_trymerge.py [hunk 1]
```diff
@@ -247,6 +247,19 @@ def test_get_author_null(self, mocked_gql: Any, *args: Any) -> None:
...
+        self.assertIsNotNone(pr.last_pushed_at())
+
+        # 307120d6d3f7fcc3f92cfd26be891d360ad6a92a is merge commit
+        # and as such does not have a pushedDate
+        # See https://github.com/pytorch/pytorch/pull/94146#issuecomment-1421647117
+        pr = GitHubPR("pytorch", "pytorch", 94146)
+        self.assertIsNone(pr.last_pushed_at())
+
     @mock.patch('trymerge.gh_graphql', side_effect=mocked_gh_graphql)
     def test_large_diff(self, mocked_gql: Any, *args: Any) -> None:
```
Comment (malfet) [inline .github/scripts/test_trymerge.py:264]: @clee2000 why have you added `*args: Any` to all test function signatures?  It's not wrong, but just curious why

**Review by huydhn — APPROVED** [review]
LGTM!

File: .github/scripts/test_trymerge.py (hunk 1 above)
Comment (clee2000) [reply .github/scripts/test_trymerge.py:264]: I added a 3 mocks at the class level and didnt want to add 3 params to all the test functions

**Pull request closed without merging**